[#64541] fixed split view resizer and content for BIM

Kharonus opened the pull request:
# Ticket
[#64541](https://community.openproject.org/work_packages/64541)

# What are you trying to accomplish?

- applied split view styles on BIM content
- removed some linting issues
- reorganized HTML structure for the resizer

Kharonus commented on frontend/src/global_styles/layout/work_packages/_details_view.sass:
```diff
@@ -29,7 +29,8 @@
 body.router--work-packages-partitioned-split-view-details,
-body.router--work-packages-partitioned-split-view-new
+body.router--work-packages-partitioned-split-view-new,
+.work-packages-partitioned-page--content-container
```
❓ @HDinger I'm not sure, why we even need the other selectors, why not only using the parent container to introduce the styles of content right below?

I wanted to add selector `body.router--bim` before, which also works, but actually I should not use selectors from the bim module in the global styles. Hence this question.

Kharonus replied:
resolved as in obsolete.

HDinger commented on frontend/src/global_styles/layout/work_packages/_details_view.sass:
```diff
@@ -29,7 +29,8 @@
 body.router--work-packages-partitioned-split-view-details,
-body.router--work-packages-partitioned-split-view-new
+body.router--work-packages-partitioned-split-view-new,
+.work-packages-partitioned-page--content-container
```
I would not add that here.. It has unwanted side effects, for example a border around the right side
<img width="1494" alt="Bildschirmfoto 2025-07-09 um 08 56 13" src="https://github.com/user-attachments/assets/8981d833-708c-455a-b0d8-6b883d982356" />

 this is even visible when the split screen is not opened at all:
<img width="1499" alt="Bildschirmfoto 2025-07-09 um 08 57 20" src="https://github.com/user-attachments/assets/43b5bcaf-0408-45ee-b2ed-bdd7e534eeb3" />

My suggestion would be, to add a rule to the `bim/../generic.sass`, given that this kind of split screen behaviour is also a bim-specific edge case.

Kharonus replied:
hmpf, figures... I didn't spot that effects. Thanks.

HDinger commented on frontend/src/app/features/work-packages/components/wp-grid/wp-grid.component.ts:
```diff
@@ -58,12 +70,6 @@ import { WorkPackageViewOutputs } from 'core-app/features/work-packages/routing/
-         class="hidden-for-mobile hide-when-print">
-      <wp-resizer [elementClass]="resizerClass"
-                  [localStorageKey]="resizerStorageKey"></wp-resizer>
-    </div>
```
If that is removed, I guess the following can be removed as well..
```
  @Input() public showResizer = false;
  @Input() public resizerClass = '';
  @Input() public resizerStorageKey = '';
  ```

Kharonus replied:
I rolled that change back. Tried to minimize the impact of this fix.

Kharonus commented:
phew, I have the feeling, there is more then something off in the structure and styling of the BCF module.

Here is my new shot at fixing things. Roughly summarize to:
- do not touch core classes or styles
- move the resizer out of table and grid, but rendering it always in the "content-right bcf list"
- fixed some paddings issues caused by that
- added some partition handling, as this seemed somehow broken completely.

HDinger commented on frontend/src/app/features/bim/ifc_models/pages/viewer/ifc-viewer-page.component.ts:
```diff
@@ -196,4 +208,20 @@ export class IFCViewerPageComponent extends PartitionedQuerySpacePageComponent i
+  }
+
+  private partitionFromDisplayRepresentation(dr:BcfViewState):ViewPartitionState {
+    if (dr === 'splitCards' || dr === 'splitTable') {
```
This is not sufficient. I cannot open the split screen of a WP any more.

Kharonus replied:
in which module? still within BCF module, right?

HDinger replied:
Yes. Before, I could open the split screen when pressing the little info icon within the card or the table.

Kharonus replied:
Okay, yeah. So, the problem is, that the click on the dropdown to change the display representation does not send a ng state transition.

the click on the info icon on the other does exactly that. But at the same time, the click on info icon does not update the query space, like the dropdown does... Wow, I have no clue, when this broke down, or what changed...

My new approach:
- transition bim.partitioned.list shows `-left-only` per default
- the display representation overwrites the partition if triggered (only on drop down click and initial load)
- info icon continues to work on transitions

Kharonus replied:
this triggers one last misbehaviour, but I'm a bit out of ideas - except reworking the whole routing in here.

- when already in split view - e.g. viewer and cards - clicking the info icon
- this opens correctly the details tab
- closing it alas now goes back to `-left-only`, only a reload gives the split screen back

HDinger commented on frontend/src/global_styles/content/modules/_bim.sass:
```diff
@@ -1,3 +1,16 @@
+    overflow: auto
+    position: relative
+    border-left: 2px solid var(--borderColor-default)
+    border-top: 2px solid var(--borderColor-default)
```
Again, you don't need that. It will create a border around the right side no matter which content you have there:
<img width="1496" alt="Bildschirmfoto 2025-07-09 um 15 55 54" src="https://github.com/user-attachments/assets/9369b59c-7026-4e3e-8832-805f786be0f7" />
<img width="1496" alt="Bildschirmfoto 2025-07-09 um 15 58 17" src="https://github.com/user-attachments/assets/0682cf10-cbad-4f9b-9912-ae3e4db47e25" />

```suggestion
body.router--bim
  @include extended-content--right

  .work-packages-partitioned-page--content-right
  // Nothing more, nothing less
    flex-basis: var(--split-screen-width) 
```

Kharonus committed:
[#64541] fixed split view resizer and content for BIM

- https://community.openproject.org/work_packages/64541
- applied split view styles on BIM content
- removed some linting issues
- reorganized HTML structure for the resizer
- fixed paddings on BIM content-right
- override resizer variables on BCF non-split content

Kharonus commented:
Ahoi @HDinger ,

can you recheck this? Like shown before, this is a more hacky, but direct approach of fixing the resizer issue.

in general I override the same variable as the resizer itself to 0, when the bcf list is not supposed to show a split view.

HDinger reviewed (commented):
I guess a comment would be nice in the code explaining why this hack is necessary. Otherwise good to merge

Kharonus committed:
[#64541] added explainatory comment for BCF split screen hack
